code/tika/TikaRestClient.php

```diff
@@ -5,20 +5,58 @@
 
 class TikaRestClient extends Client
 {
+    /*
+    * @var Array
+    */
+    private static $_options = ['username' => null, 'password' => null];
+
+    /*
+    * @var Boolean
+    */
+    private static $_available = false;
+
+    /*
+    * @var Float
+    */
+    private static $_version = 0.0;
+
+    /*
+    * @var Array
+    */
+    protected $mimes = [];
+
+    public function __construct($baseUrl = '', $config = null)
+    {
+        if (defined('SS_TIKA_USERNAME') && defined('SS_TIKA_PASSWORD')) {
+            self::$_options = [
+                'username' => SS_TIKA_USERNAME,
+                'password' => SS_TIKA_PASSWORD,
+            ];
+        }
+        parent::__construct($baseUrl, $config);
+    }
+
     /**
      * Detect if the service is available
      *
      * @return bool
      */
     public function isAvailable()
     {
-        try {
-            return $this
-                ->get()->send()
-                ->getStatusCode() == 200;
-        } catch (RequestException $ex) {
-            return false;
+        if (!self::$_available) {
```

Comment:
Does this need to be static? If not can we use a setter/getter with a config backing? I think it would be better to use this class as a singleton than to implement singleton like properties inside it, especially if they look the same as config properties